src/coreclr/inc/corinfo.h

```diff
@@ -3194,7 +3194,8 @@ class ICorDynamicInfo : public ICorStaticInfo
     // return the data's address (for static fields only)
     virtual void* getFieldAddress(
                     CORINFO_FIELD_HANDLE    field,
-                    void                  **ppIndirection = NULL
+                    void                  **ppIndirection = NULL,
+                    CORINFO_OBJECT_HANDLE  *pFrozenBoxedStatic = NULL
```

Comment:
We have `CORINFO_FLG_FIELD_STATIC_IN_HEAP` for this already. This extra argument looks redundant with it. Can we just not set `CORINFO_FLG_FIELD_STATIC_IN_HEAP` for statics allocated in frozen heap?

Reply:
Ah, yep, just merged a version with only `CORINFO_FLG_FIELD_STATIC_IN_HEAP` @SingleAccretion suggested the same in Discord 🙂

Reply:
as a bonus it now emits better codegen since it no longer need to use addressing mode on top of frozen handle.